doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1257,409 +1261,344 @@ Managing the local context
-   :n:`pose proof @term as @ident`, following the model of :n:`pose
-   (@ident := @term)` but dropping the value of :token:`ident`.
+   .. tacn:: eenough ( @ident : @type ) {? by @ltac_expr3 }
+             eenough @one_term {? @as_ipat } {? by @ltac_expr3 }
+      :name: eenough; _
 
-.. tacv:: epose proof (@ident := @term)
+      Unlike :tacn:`enough`, the :n:`@type`\s and :n:`@one_term`\s in
+      :tacn:`eenough` may contain :term:`holes <hole>`, denoted by :n:`_`,
+      for which the tactic will create existential variables.  This lets you
+      avoid specifying the asserted statement completely until you later
+      start to prove it.
 
-   This is an alternative syntax for :n:`eassert (@ident := @term)`
-   and :n:`epose proof @term as @ident`, following the model of
-   :n:`epose (@ident := @term)` but dropping the value of
-   :token:`ident`.
+.. tacn:: cut @one_term
 
-.. tacv:: enough (@ident : @type)
-   :name: enough
+   Implements the non-dependent case of the :ref:`App <app_rule>` typing rule,
+   the Modus Ponens inference rule.  It is equivalent to
+   :n:`enough (@ident: @one_term). revert @ident.`
+   This tactic is generally considered obsolete but it is still widely
+   used in old scripts.
 
-   This adds a new hypothesis of name :token:`ident` asserting :token:`type` to the
-   goal the tactic :tacn:`enough` is applied to. A new subgoal stating :token:`type` is
-   inserted after the initial goal rather than before it as :tacn:`assert` would do.
+.. tacn:: pose proof @term {? @as_ipat }
+          pose proof ( @ident := @term )
+   :name: pose proof; _
 
-.. tacv:: enough @type
+   The first form behaves like :n:`assert @type {? @as_ipat } by exact @term`
+   where :token:`type` is the type of :token:`term`.
 
-   This behaves like :n:`enough (@ident : @type)` with the name :token:`ident` of
-   the hypothesis generated by Coq.
+   .. Théo notes it's odd that the first form uses @term instead of @one_term
 
-.. tacv:: enough @type as @simple_intropattern
+   The second form is equivalent to :n:`assert (@ident := @term)`.
 
-   This behaves like :n:`enough @type` using :token:`simple_intropattern` to name or
-   destruct the new hypothesis.
+   .. tacn:: epose proof @term {? @as_ipat }
+             epose proof ( @ident := @term )
+      :name: epose proof; _
 
-.. tacv:: enough (@ident : @type) by @tactic
-          enough @type {? as @simple_intropattern } by @tactic
+      While :tacn:`pose proof` expects that no existential variables are generated by
+      the tactic, :tacn:`epose proof` removes this constraint.
 
-   This behaves as above but with :token:`tactic` expected to solve the initial goal
-   after the extra assumption :token:`type` is added and possibly destructed. If the
-   :n:`as @simple_intropattern` clause generates more than one subgoal, :token:`tactic` is
-   applied to all of them.
+.. tacn:: specialize @one_term_with_bindings {? as @simple_intropattern }
```

Comment:
Should use `@as_ipat`, since it is equivalent.

Reply:
I'll change this one.  `{? as @simple_intropattern }` also appears in the `functional induction`, `inversion_sigma` and `soft functional induction`.  It makes sense to standardize such common elements when there's useful description at the common point (`@as_ipat` in this case).  Otherwise, it seems a bit more readable to skip the extra indirection and make immediately apparent that `as` is a literal value.  OTOH, doing this systematically for a bunch of these could be quite a few edits in `common.edit_mlg`, though maybe the mlgs could be refactored for consistency.

Reply:
Yes, I agree with everything you said.

FTR, when I initially saw `as_ipat` in this chapter, I first considered suggesting splicing it, but then I noticed that it was so commonly used that it made sense to have a common description.